### PR TITLE
[script] [hunting-buddy] mitigate thinking room is taken when someone walks by

### PR DESCRIPTION
**Pull request by KatoakDR**
### Background
* Fixes https://github.com/rpherbig/dr-scripts/issues/2444

### Changes
* When `hunting-buddy` has found a room and is waiting ~10 seconds to see if the room is available, don't immediately leave just because someone else walked into or through the room. Instead, look for more telling signs like the person engages a critter or attacks.

### Tests
* Recruiting testers in [Discord](https://discord.com/channels/745675889622384681/745675890242879671/947707157850247218)
* I've tested that the logic will move on if someone shows up and attacks in the room; but will claim the room if someone walks through or stands in the room but doesn't attack

### CODE REVIEW

File: hunting-buddy.lic [hunk 1]
```diff
@@ -324,12 +334,14 @@ class HuntingBuddy
                 return UserVars.friends.find { |friend| Flags["hunting-buddy-room-check-#{friend}"] }
               end
               fput("say #{@settings.empty_hunting_room_messages.sample}") unless @settings.empty_hunting_room_messages.empty?
+              room_is_empty = true
               20.times do |_|
                 pause 0.5
                 return true if UserVars.friends.find { |friend| Flags["hunting-buddy-room-check-#{friend}"] }
-                return false if Flags['hunting-buddy-room-check'] || !(DRRoom.pcs - DRRoom.group_members - UserVars.friends).empty?
+                return false if Flags['hunting-buddy-room-check']
+                room_is_empty = !(DRRoom.pcs - DRRoom.group_members - UserVars.friends).empty?
```
[the remark this answers is not in the document]
Comment (asechrest) [reply hunting-buddy.lic:342]: PS - Thanks for the PR!

[the remark this answers is not in the document]
Comment (KatoakDR) [reply hunting-buddy.lic:342]:
```suggestion
                room_is_empty = (DRRoom.pcs - DRRoom.group_members - UserVars.friends).empty?
```

[the remark this answers is not in the document]
Comment (KatoakDR) [reply hunting-buddy.lic]: Good catch, thanks

**Comment by MahtraDR** [comment]
Webhook test.

**Comment by asechrest** [comment]
Since the mid-day commit I've been running this fine. @MahtraDR ?

**Comment by KatoakDR** [comment]
> Since the mid-day commit I've been running this fine. @MahtraDR ?

Thanks for the follow up

**Comment by MahtraDR** [comment]
> Since the mid-day commit I've been running this fine. @MahtraDR ?

I was having computer problems today. I've got it going on mine now.

**Review by MahtraDR — APPROVED** [review]
Looks good on my end.

**Pull request merged**